tests/trainer/resources/rhai_speculator.ipynb

```
@@ -0,0 +1,645 @@
+    "                fs.get(remote_file, local_file)\n",
+    "                count += 1\n",
+    "            print(f\"  Downloaded {count} files to {model_local_path}\")\n",
+    "        else:\n",
+    "            print(f\"  Model already exists at {model_local_path}, skipping\")\n",
+    "\n",
+    "        print(\"S3 download complete\")\n",
+    "    else:\n",
+    "        print(\"HuggingFace mode: downloading model from HF Hub\")\n",
+    "        if not os.path.exists(os.path.join(model_local_path, \"config.json\")):\n",
+    "            from huggingface_hub import snapshot_download\n",
+    "            snapshot_download(\n",
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)